polynote-frontend/polynote/messaging/dispatcher.ts

```diff
@@ -44,1189 +50,335 @@ export class NotebookMessageDispatcher extends MessageDispatcher<NotebookState,
-        super();
-        Object.freeze(this);
-    }
-
-    static unapply(inst: RequestCompletions): ConstructorParameters<typeof RequestCompletions> {
-        return [inst.cellId, inst.offset, inst.resolve, inst.reject];
-    }
-}
-
-export class RequestSignature extends UIAction {
-    constructor(readonly cellId: number, readonly offset: number,
-                readonly resolve: (value: SignatureHint) => void, readonly reject: () => void) {
-        super();
-        Object.freeze(this);
-    }
-
-    static unapply(inst: RequestSignature): ConstructorParameters<typeof RequestSignature> {
-        return [inst.cellId, inst.offset, inst.resolve, inst.reject];
-    }
-}
-
-export class RequestNotebookVersion extends UIAction {
-    constructor(readonly version: number) {
-        super();
-        Object.freeze(this);
-    }
-
-    static unapply(inst: RequestNotebookVersion): ConstructorParameters<typeof RequestNotebookVersion> {
-        return [inst.version];
-    }
-}
-
-export class RequestCellRun extends UIAction {
-    constructor(readonly cells: number[]) {
-        super();
-        Object.freeze(this);
-    }
-
-    static unapply(inst: RequestCellRun): ConstructorParameters<typeof RequestCellRun> {
-        return [inst.cells];
-    }
-}
-
-export class RemoveTask extends UIAction {
-    constructor(readonly taskId: string) {
-        super();
-        Object.freeze(this);
-    }
-
-    static unapply(inst: RemoveTask): ConstructorParameters<typeof RemoveTask> {
-        return [inst.taskId];
-    }
-}
-
-export class RemoveError extends UIAction {
-    constructor(readonly err: ServerErrorWithCause) {
-        super();
-        Object.freeze(this);
-    }
-
-    static unapply(inst: RemoveError): ConstructorParameters<typeof RemoveError> {
-        return [inst.err];
-    }
-}
-
-export class RequestCancelTasks extends UIAction {
-    constructor() {
-        super();
-        Object.freeze(this);
-    }
-
-    static unapply(inst: RequestCancelTasks): ConstructorParameters<typeof RequestCancelTasks> {
-        return [];
-    }
-}
+        cellView.view("currentSelection").addObserver(range => {
+            if (range) {
+                this.socket.send(new messages.CurrentSelection(id, range))
+            }
+        }, this)
 
-export class RequestClearOutput extends UIAction {
-    constructor() {
-        super();
-        Object.freeze(this);
     }
 
-    static unapply(inst: RequestClearOutput): ConstructorParameters<typeof RequestClearOutput> {
-        return [];
+    private sendUpdate(upd: NotebookUpdate) {
+        this.socket.send(upd)
+        ClientBackup.updateNb(this.handler.state.path, upd)
+            .catch(err => console.error("Error backing up update", err))
     }
-}
 
-export class RequestNotebooksList extends UIAction {
-    constructor() {
-        super();
-        Object.freeze(this);
-    }
+    /*******************************
```

Comment:
The methods below (and equivalent in `ServerMessageDispatcher`, below), are basically the 
same changes as in https://github.com/polynote/polynote/pull/986